bzzeth/peer.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package bzzeth
+
+import (
+	"math/rand"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethersphere/swarm/p2p/protocols"
+)
+
+// Peer extends p2p/protocols Peer and represents a conrete protocol connection
+type Peer struct {
+	*protocols.Peer            // embeds protocols.Peer
+	serveHeaders    bool       // if the remote serves headers
+	requests        *requests  // per-peer pool of open requests
```

Comment:
i'm pretty sure that pointer is not needed here

Reply:
The same argument as *peers apply here. 
If we change.. we need to change both...

Reply:
please explain how does the comment about `*peers` relates to this comment

Reply:
requests is like peer... a struct which has a lock and a map.
A request object is created and added to it..
similar to *peer

Reply:
The pointer is needed. If is not a pointer, then the mutex from requests will be copied. Also, requests type is using pointer semantics which is consistent here in Peer field.